dags/gcp_spy.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+
+import requests
+
+from airflow import utils, DAG
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.email_operator import EmailOperator
+from airflow.operators.python_operator import PythonOperator
+
+gcp_services = [
+    ('sql', 'Cloud SQL'),
+    ('spanner', 'Spanner'),
+    ('bigtable', 'BigTable'),
+    ('compute', 'Compute Engine'),
+]
+
+INSTANCES_IN_PROJECT_TITLE = "Instances in `polidea-airflow`:"
+
+
+def send_slack_msg(**context):
+    with_instances = []
+    with_instances_label = INSTANCES_IN_PROJECT_TITLE
+    no_instances = []
+
+    for gcp_service in gcp_services:
+        result = context['task_instance'].\
+            xcom_pull(task_ids='gcp_service_list_instances_{}'.format(gcp_service[0]))
+        print("result: {}".format(result))
```

Comment:
Maybe use logging :). Not super necessary but haters gonna hate

Reply:
One more comment. I think it will be better to put all the code in a class, rather than as top-level functions. That might make logging easier (extends LoggingMixin) and we will have the opportunity to show LoggingMixin as one of the Airflow built-in utils. You can as easily pass class methods as top-level methods as callables.

Reply:
Good idea, will refactor tomorrow morning 👍

Reply:
Logging abandoned, as discussed offline